Fix Array Knob deserialization

rylnd opened the pull request:
Issue: Array Knob data retrieved from the URL is stored as an Object

## What I did
Updated the Array knob's `deserialize` method to handle Objects.

## How to test
Without these changes, if you reload the page with a component using the `array` knob and expecting an Array; you'll get the error
```
warning.js:33 Warning: Failed prop type: Invalid prop `knob.value` of type `object` supplied to `ArrayType`, expected `array`.
```

### Is this testable with jest or storyshots?
Yes; unit tests have been added for Array's `deserialize`.

### Does this need a new example in the kitchen sink apps?
No.

### Does this need an update to the documentation?
No.

Commit:
Rename import to avoid naming conflict

Commit:
Allow Array to be deserialized from an Object

When an Array knob's value is serialized from the URL into the store, it
is an Object.

This ensures that when a user refreshes the page, the Array knob will
properly deserialize the Object into an Array.

danielduan commented:
Thanks for the PR and the test!

Hypnosphi commented on addons/knobs/src/components/types/Array.js:
```diff
@@ -47,6 +47,10 @@ ArrayType.propTypes = {
+ArrayType.deserialize = value => {
+  if (Array.isArray(value)) return value;
+
+  return Object.keys(value).reduce((array, key) => [...array, value[key]], []);
```
Does this rely on `Object.keys` being sorted? It's actually not guaranteed to be:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/keys
> The Object.keys() method returns an array of a given object's own enumerable properties, in the same order as that provided by a for...in loop

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for...in#Array_iteration_and_for...in
> There is no guarantee that for...in will return the indexes in any particular order

Please sort the keys before reducing

rylnd replied:
@Hypnosphi as evidenced in the unit test, the order of keys does not matter.

rylnd replied:
@Hypnosphi 2f22c2c

Hypnosphi committed:
Merge branch 'master' into fix-array-deserialize

rylnd committed:
Ensure deserialized Array is ordered by index

Object.keys does not guarantee any particular order, which could result
in an out-of-order array if we did not sort said keys.

rylnd committed:
Remove redundant expectations

The remaining expectations implicitly check whether the objects are
arrays.

Hypnosphi committed:
Merge branch 'master' into fix-array-deserialize
